packages/web/src/common/store/cache/tracks/sagas.ts

```diff
@@ -118,22 +123,18 @@ function* editTrackAsync(action: ReturnType<typeof trackActions.editTrack>) {
     )
   }
 
-  const [{ metadata: trackForEdit }] = yield* processTracksForUpload([
-    // TODO: this is ok because processTracksForUpload only uses the metadata
-    // but ideally we update it to accept TrackMetadata[] instead
-    { metadata: action.formFields } as TrackForUpload
-  ])
+  const trackForEdit = yield* processTrackForUpload(action.formFields)
 
   yield* call(
     confirmEditTrack,
     action.trackId,
-    trackForEdit as ExtendedTrackMetadata,
+    trackForEdit,
     wasUnlisted,
     isNowListed,
     currentTrack
   )
 
-  const track = { ...trackForEdit } as ExtendedTrackMetadata & Track
+  const track = { ...trackForEdit } as Track & TrackMetadataForUpload
```

Comment:
no unfortunately - we're actually extending that type to add cover art sizes below. The reason this cast is necessary is because we're copying the object and making a new object that's a hybrid... the cache is a liar and not all types there are the same as what you'd expect